tests/gold_tests/headers/general-connection-failure-502.test.py

```diff
@@ -38,7 +38,8 @@
 
 tr = Test.AddTestRun()
 tr.Processes.Default.StartBefore(Test.Processes.ts)
-tr.Processes.Default.Command = "python tcp_client.py 127.0.0.1 {0} {1} | egrep -v '^(Date: |Server: ATS/)'"\
+# Do not start the origin server: We wish to simulate connection refused while hopefully no one else uses this port.
+tr.Processes.Default.Command = "python tcp_client.py 127.0.0.1 {0} {1} | sed -e '/^Date: /d' -e '/^Server: ATS\//d'"\
```

Comment:
The code change here is to use `sed`—a stream-based editor—rather than `grep`—a line-based matcher—because the latter will add a line-feed character to the end of the output, which is not what we want to test.